demuxer.go

```diff
@@ -40,10 +40,10 @@ func NewDemuxer(ctx context.Context, r io.Reader, opts ...func(*Demuxer)) (d *De
 	// Init
 	d = &Demuxer{
 		ctx:        ctx,
-		packetPool: newPacketPool(),
 		programMap: newProgramMap(),
 		r:          r,
 	}
+	d.packetPool = newPacketPool(d)
```

Comment:
It feels weird to inject the demuxer in the packet pool 🤔 

I'd rather move the creation of the packet pool after applying options and providing `d.optPacketsParser` and `d.programMap` to the packet pool.

Then it can call the global `parseData` function

Reply:
Yea agreed this feels a little hacky..

I wonder if access to the demuxer is even needed if we're not planning to use parseData long term.

I'll see if I can improve the accumulator to check size directly and re-evaluate.